### PR TITLE
refactor: Simplify nested registry iteration

### PR DESCRIPTION
**Pull request by morgsmccauley**
In Coordinator, to iterate over all Indexers you need two loops: one for the accounts, and another for its functions. As iteration is quite common, I've added a custom `Iterator` implementation which achieves the same with a single loop.

### CODE REVIEW

File: coordinator/src/indexer_state.rs [hunk 1]
```diff
@@ -87,18 +87,17 @@ impl IndexerStateManagerImpl {
...
 
-                if indexer_state.enabled {
-                    filtered_registry
-                        .entry(account_id.clone())
-                        .or_default()
-                        .insert(function_name.clone(), indexer_config.clone());
-                }
+            if indexer_state.enabled {
+                filtered_registry
+                    .0
```
Comment (darunrs) [inline coordinator/src/indexer_state.rs:97]: What is this `.0`? Is this another way of getting the first item in an array? Curious why it's an array anyway if so.

File: coordinator/src/registry.rs [hunk 2]
```diff
@@ -13,7 +14,58 @@ use registry_types::AllIndexers;
...
+
+    pub fn iter(&self) -> IndexerRegistryIter {
+        IndexerRegistryIter {
+            account_iter: self.0.iter(),
+            function_iter: None,
+        }
+    }
+}
+
+pub struct IndexerRegistryIter<'a> {
```
Comment (darunrs) [inline coordinator/src/registry.rs:37]: Is the position of the iterator stateful? Like, if I looped through the registry, and then broke the loop half way. And then tried to iterate again.  Would it reset or pick up where I left off?

**Review by darunrs — APPROVED** [review]
Cool refactoring!

File: coordinator/src/indexer_state.rs (hunk 1 above)
Comment (morgsmccauley) [reply coordinator/src/indexer_state.rs:97]: `IndexerRegistry` is a "Tuple Struct", `.0` is how you access elements within a tuple

File: coordinator/src/registry.rs (hunk 2 above)
Comment (morgsmccauley) [reply coordinator/src/registry.rs:37]: The iterator itself is stateful, yes. That would be possible, but you'd need to store the iterator outside the look, i.e.
```rust
const iter = registry.iter();

for i in iter {
  // break
}

for j in iter {
}
```

If you called `.iter()` twice you'd get two separate stateful instances.

**Pull request merged**